fdbclient/BackupAgentBase.actor.cpp

```diff
@@ -321,6 +327,19 @@ ACTOR static Future<Void> decodeBackupLogValue(Arena* arena,
 				logValue = logValue.decrypt(cipherKeys, tempArena, BlobCipherMetrics::BACKUP);
 			}
 			ASSERT(!logValue.isEncrypted());
+
+			if (config.tenantMode == TenantMode::REQUIRED && !isSystemKey(logValue.param1)) {
```

Comment:
I think there is one potential gotcha here, which is that right after a recovery before the cluster has recovered the configuration, the cluster will initially be in disabled tenant mode. The thing that ordinarily saves us in that case is that you won't be able to commit anything (unless you have a special transaction that allows provisional proxies), so if you are running a transaction your commit would fail and you would retry.

We need to evaluate what happens here in that case to make sure that we properly handle the recovery case.

Reply:
This code runs in the commit proxy so (applyMetadataMutations) so not sure if we can run into the problem where the configuration is outdated (unless this code can get called from the provisional proxies)?

Reply:
Furthermore this case only runs when we are restoring non-system keys. So by this point im assuming the conf system keys have been restored?